rust_src/src/process.rs

```diff
@@ -265,15 +265,11 @@ pub fn set_process_plist(mut process: LispProcessRef, plist: LispObject) -> Lisp
 /// nil, indicating the current buffer's process.
 #[lisp_fn]
 pub fn process_status(process: LispObject) -> LispObject {
-    let p = if process.is_string() {
-        get_process(process)
-    } else {
-        unsafe { cget_process(process) }
+    let mut process = match get_process(process) {
+        Qnil => return process,
+        p => p.as_process_or_error(),
```

Comment:
@nickdrozd has been working to remove these `or_error` calls and replacing them with `.into()` instead.

Reply:
Ah, right.

Reply:
We need to update the wiki pages...

Reply:
I opened #1253

Reply:
I never liked this _or_error naming.

Reply:
pros and cons. `as_foo_or_error()` lets me know it will die if it fails. `foo.into()` is now sometimes safe and sometimes not depending on whether it returns an Option or if there is a fall back. But yeah, the code is moving towards a little less verbosity which is probably good.